### PR TITLE
#228: first cut at supporting Pandas 2.0+

### PR DESCRIPTION
**Pull request by DavidRothblum**
Addresses the necessary changes when upgrading to Pandas 2.0+. 

The farthest reaching breaking change appears to be in `_check_is_datetime_type`, which has been changed to use the new `pd.api.types` methods. 

A lot of the changes here address FutureWarnings or DeprecationWarnings: 
    - `‘H’` and `’T’` frequency strings are deprecated -> Use `‘h’` or `‘min’` 
    - Passing bytes to `pd.read_excel` is deprecated -> Wrap with `io.BytesIO`
    - `DataFrame.applymap` deprecated -> `Use DataFrame.map`
    - Pandas/Tabula error=“ignore” deprecated -> Handle `pd.to_numeric` errors explicitly
    - Pandas set on copy of data frame -> replace `.fillna(_, inplace=True)` reference

### CODE REVIEW

**Comment by DavidRothblum** [comment]
Fixes error reported in #227: 
```
In [1]: import gridstatus

In [2]: import pandas as pd

In [3]: pd.__version__
Out[3]: '2.2.1'

In [4]: iso = gridstatus.Ercot()

In [5]: iso.get_rtm_spp(2023)
Out[5]: 
                            Time            Interval Start              Interval End    Location Location Type            Market    SPP
0      2023-01-01 00:00:00-06:00 2023-01-01 00:00:00-06:00 2023-01-01 00:15:00-06:00   HB_BUSAVG   Trading Hub  REAL_TIME_15_MIN  -2.56
1      2023-01-01 00:00:00-06:00 2023-01-01 00:00:00-06:00 2023-01-01 00:15:00-06:00  HB_HOUSTON   Trading Hub  REAL_TIME_15_MIN  -2.56
2      2023-01-01 00:00:00-06:00 2023-01-01 00:00:00-06:00 2023-01-01 00:15:00-06:00    LZ_RAYBN     Load Zone  REAL_TIME_15_MIN  -2.56
3      2023-01-01 00:00:00-06:00 2023-01-01 00:00:00-06:00 2023-01-01 00:15:00-06:00     LZ_WEST     Load Zone  REAL_TIME_15_MIN  -2.56
4      2023-01-01 00:00:00-06:00 2023-01-01 00:00:00-06:00 2023-01-01 00:15:00-06:00   HB_HUBAVG   Trading Hub  REAL_TIME_15_MIN  -2.56
...                          ...                       ...                       ...         ...           ...               ...    ...
805915 2023-12-31 23:45:00-06:00 2023-12-31 23:45:00-06:00 2024-01-01 00:00:00-06:00    LZ_SOUTH     Load Zone  REAL_TIME_15_MIN   4.66
805916 2023-12-31 23:45:00-06:00 2023-12-31 23:45:00-06:00 2024-01-01 00:00:00-06:00     LZ_WEST     Load Zone  REAL_TIME_15_MIN  15.32
805917 2023-12-31 23:45:00-06:00 2023-12-31 23:45:00-06:00 2024-01-01 00:00:00-06:00      LZ_AEN     Load Zone  REAL_TIME_15_MIN  13.72
805918 2023-12-31 23:45:00-06:00 2023-12-31 23:45:00-06:00 2024-01-01 00:00:00-06:00    LZ_RAYBN     Load Zone  REAL_TIME_15_MIN  13.92
805919 2023-12-31 23:45:00-06:00 2023-12-31 23:45:00-06:00 2024-01-01 00:00:00-06:00     LZ_WEST     Load Zone  REAL_TIME_15_MIN  15.32

[805920 rows x 7 columns]
```

**Comment by DavidRothblum** [comment]
Looks like the docs are failing to build. I don't have much familiarity with readthedocs/sphinx, so would appreciate some tips on how to fix!

File: gridstatus/ercot.py [hunk 1]
```diff
@@ -328,15 +328,13 @@ def get_fuel_mix(self, date, verbose=False):
...
         for day in data["data"].keys():
-            df = (
-                pd.DataFrame(data["data"][day])
-                .applymap(
-                    lambda x: x["gen"],
-                    na_action="ignore",
+            df = pd.DataFrame(data["data"][day])
+            df_transformed = df.apply(
+                lambda col: col.apply(
+                    lambda x: x.get('gen') if isinstance(x, dict) else pd.NA
```
Comment (WillKoehrsen) [inline gridstatus/ercot.py:334]: The ruff linter is complaining this needs to be `"gen"`.

**Comment by WillKoehrsen** [comment]
> Looks like the docs are failing to build, but work locally with `make docs`. I don't have much familiarity with readthedocs/sphinx, so would appreciate some tips on how to fix!

I don't entirely understand the readthedocs build process. The GitHub action that builds the docs did succeed.

The [output from the readthedocs build](https://readthedocs.org/projects/isodata/builds/23570913/) says 

```
Could not import extension myst_nb (exception: No module named 'myst_nb')
```

but `myst-nb` should be installed since it's a requirement in `pyproject.toml`.

File: gridstatus/ercot.py (hunk 1 above)
Comment (DavidRothblum) [reply gridstatus/ercot.py]: 🤦 ... I see you guys use black formatting 🎉 . Fixed the formatting with black on the latest commit!

**Comment by WillKoehrsen** [comment]
The readthedocs (RTD from now on) issue may be caused by RTD not properly using `poetry`. [Here's a comment](https://github.com/readthedocs/readthedocs.org/issues/4912#issuecomment-1110189914) that shows how to update `.readthedocs.yaml` to use `poetry`.

**Comment by WillKoehrsen** [comment]
I fixed the RTD build issue in another PR: #346. If you want, feel free to move those changes into this PR and I'll close that one.

**Review by WillKoehrsen — APPROVED** [review]
Looks good, thanks a lot for taking this on!

**Pull request merged**